ContactMechanics/Systems.py

```diff
@@ -485,6 +487,11 @@ def fun(disp):
 
         return fun
 
+    def hessp(self, disp):
+        prod = -self.substrate.evaluate_force(
+            disp.reshape(self.substrate.nb_domain_grid_pts))
+        return prod.reshape(-1)
```

Comment:
prod.reshape(disp.shape) 
Then the shape of output is always consistent with rthe shape of the input.